packages/gatsby/lib/redux/reducers/page-components.js

```diff
@@ -0,0 +1,20 @@
+module.exports = (state = {}, action) => {
+  switch (action.type) {
+    case "ADD_PAGE_COMPONENT":
+      if (!state[action.payload.componentPath]) {
```

Comment:
This kind of pattern affraid me (a bad past experience this `object[a.b.c]`…) It's proofer to use the lodash `get` (it's will not break if you have an `undefined` somewhere 😃 )

So for this => `get(state, 'action.payload.componentPath')`

(I know, normally we never should have `undefined` in `payload`, it's just a protection, it's not a big cost and really improve an application stability)